datadog_checks_base/datadog_checks/base/utils/db/statement_metrics.py

```diff
@@ -23,6 +23,7 @@ class StatementMetrics:
 
     def __init__(self):
         self._previous_statements = {}
+        self._cache_initialized = False
```

Comment:
nit: could we get rid of this var and set `self._previous_statements = None` here? I think that's cleaner, but possibly less clear.

Reply:
I like this suggestion and made that change.